### ISSUE TITLE
rocketchatctl update results in line 909: [: 3.15.0: unary operator expected

### ISSUE THREAD

**Issue by patrickbussmann** [RocketChat/install.sh#42]
I installed RocketChat.
I installed RocketChat CTL.
And then I tried to run `/usr/local/bin/rocketchatctl update`

Which results in:

```
Updates could be risky, you can use the backup option # rocketchatctl backup, to create a backup of the rocketchat database first. Please check that you have enough space in your system to store the backup.
/usr/local/bin/rocketchatctl: line 909: [: 3.15.0: unary operator expected
Could not determine if updates available for RocketChat server.
```

When I manually set the version to 3.14.1 it works with the update.
It seems that the version get command is broken.

**Comment by santalex** [comment]
Hi, try this #43.

**Comment by m4z** [comment]
Probably duplicate of #34 and fixed by #40.

**Comment by franckadil** [comment]
Here we are in 2023, same issue curl -L https://releases.rocket.chat/latest/download -o /tmp/rocket.chat.tgz does pull an HTML encoded file instead of the archive.

Rocketchatctl can't find the last version.

**Comment by m4z** [comment]
What you're describing doesn't have anything to do with this issue, as far as I can tell. The problem you describe is completely different from the problem described above.

Do you even see the error messages from above? How do you go from there to a download / curl problem? If you see the above error, the download doesn't even start.

**Comment by franckadil** [comment]
Hi @m4z, I spent **few hours troubleshooting**, I should have given context since many people will see this, but for me it is again a recurring issue that happens every major upgrade (since 4 years), hence my frustration. 

**Now let dive to where my statement is coming from:**
If you read closely the entire prompt message you'll see: `Could not determine if updates available for RocketChat server.`

So the the first logical step is to check if Rocket.Chat installation works [Manual installation](https://docs.rocket.chat/deploy/prepare-for-your-deployment/other-deployment-methods/manual-installation/debian-based-distros/ubuntu) witch doesn't see #74 ,  [download link broken](https://github.com/RocketChat/Rocket.Chat/issues/29769)  etc...

So when when running the curl to pull the latest file, a file of 21 bytes is pulled looks like the server is having issues:

```
curl -L https://releases.rocket.chat/latest/download -o /tmp/rocket.chat.tgz
% Total % Received % Xferd Average Speed Time Time Time Current
Dload Upload Total Spent Left Speed
100 21 100 21 0 0 123 0 --:--:-- --:--:-- --:--:-- 125

```
**The file is only 21 bytes**


**Running a wget get confirms a RocketChat service issue:**

```
wget https://releases.rocket.chat/latest/download
--2023-07-08 17:10:43-- https://releases.rocket.chat/latest/download
Resolving releases.rocket.chat (releases.rocket.chat)... 35.196.39.85
Connecting to releases.rocket.chat (releases.rocket.chat)|35.196.39.85|:443... connected.
HTTP request sent, awaiting response... 500 Internal Server Error
2023-07-08 17:10:43 ERROR 500: Internal Server Error.
```
Unpacking the archive with `tar -xzf /tmp/rocket.chat.tgz -C /tmp` results in:

```
gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
```
**Trying  file rocket.chat.tgz displays:**

`rocket.chat.tgz: HTML document, UTF-8 Unicode text`

Now if you spend some time on google researching, you'll find multiple users have mentioned this issue and every time the Rocket.Chat team had either a web service issue or Github repository links broken. 

I should also mention that their rocketchatctl is outdated but that's an other story.

**Comment by m4z** [comment]
> **Now let dive to where my statement is coming from:** If you read closely the entire prompt message you'll see: `Could not determine if updates available for RocketChat server.`
> 
> So the the first logical step is to check if Rocket.Chat installation works [Manual installation](https://docs.rocket.chat/deploy/prepare-for-your-deployment/other-deployment-methods/manual-installation/debian-based-distros/ubuntu) witch doesn't see #74 , [download link broken](https://github.com/RocketChat/Rocket.Chat/issues/29769) etc...

I disagree. This repo is about rocketchatctl. What happens during manual install has nothing to do with that. For me the next logical step was to understand and hotfix rocketchatctl so I could propose a [fix](https://github.com/RocketChat/install.sh/issues/34).

Had I encountered more problems then, both your and my next steps wouldn't have much to do with this issue, they'd be *separate* issues and should be handled separately to keep good bug hygiene and have concise, easy-to-fix issues.
Also, the error reported in this issue was fatal. There can't be another error after that.
Also also, back when #34 and this issue were reported, the download link wasn't a problem, so your problem definitely doesn't have anything to do with this issue other than the preceding error message might be the same.

I share your frustration about these and other issues, but making issues harder to fix by compounding unrelated issues won't help anybody.

 
> Now if you spend some time on google researching, you'll find multiple users have mentioned this issue and every time the Rocket.Chat team had either a web service issue or Github repository links broken.

Which has nothing to do with this issue. This issue is about a code bug in rocketchatctl.


> I should also mention that their rocketchatctl is outdated but that's an other story.

No wonder, since it seems it (silently) fell out of official support a while ago.
